src/cargo/util/toml_mut/dependency.rs

```diff
@@ -921,6 +921,47 @@ impl Display for WorkspaceSource {
     }
 }
 
+impl Dependency {
+    /// Convert denpency as feature to TOML.
+    /// Panics if the path is relative
+    pub fn to_toml_as_feature(&self, crate_root: &Path) -> toml_edit::Item {
```

Comment:
By putting this here it and `pub`, it over-generalizes this especially when we already have `as_feature`.  I think it'd be cleaner and easier to read if this was inlined into the caller